### PR TITLE
Continue execution of `gh gei reclaim-mannequin --csv` if a username doesn't exist

### PR DESCRIPTION
**Pull request by timrogers**
Since https://github.com/github/gh-gei/pull/868, the `gh gei reclaim-mannequin` command with the `--csv` option now exits if it is given a username that doesn't exist, rather than printing a warning and continuing to the next mapping in the CSV.

This fixes that behaviour, handling the `OctoshiftCliException` thrown by `GithubApi` and printing a warning instead of exiting.

Fixes https://github.com/github/gh-gei/issues/897.

- [x] Did you write/update appropriate tests
- [x] Release notes updated (if appropriate)
- [x] Appropriate logging output
- [x] Issue linked
- [ ] Docs updated (or issue created)
- [ ] New package licenses are added to `ThirdPartyNotices.txt` (if applicable)

### CODE REVIEW

File: src/OctoshiftCLI.Tests/Octoshift/ReclaimServiceTests.cs [hunk 1]
```diff
@@ -437,7 +437,7 @@ public async Task ReclaimMannequins_NoTarget_No_Reclaim_IssuesError()
 
             _mockGithubApi.Setup(x => x.GetOrganizationId(TARGET_ORG).Result).Returns(ORG_ID);
             _mockGithubApi.Setup(x => x.GetMannequins(ORG_ID).Result).Returns(mannequinsResponse);
-            _mockGithubApi.Setup(x => x.GetUserId(TARGET_USER_LOGIN)).Returns(Task.FromResult<string>(null));
+            _mockGithubApi.Setup(x => x.GetUserId(TARGET_USER_LOGIN)).Throws(new OctoshiftCliException($"Could not resolve to a User with the login of '{MANNEQUIN_LOGIN}'."));
```
Comment (timrogers) [inline src/OctoshiftCLI.Tests/Octoshift/ReclaimServiceTests.cs:440]: This unit test wasn't realistic before - it didn't reflect the actual behaviour of `GetUserId`.

File: src/Octoshift/ReclaimService.cs [hunk 2]
```diff
@@ -162,9 +162,13 @@ public virtual async Task ReclaimMannequins(string[] lines, string githubTargetO
                     continue;
                 }
 
-                var claimantId = await _githubApi.GetUserId(claimantLogin);
+                string claimantId;
```
Comment (timrogers) [inline src/Octoshift/ReclaimService.cs:165]: I think this is the best way to deal with the scoping of the `try` block - but I'm not a C# expert, so do feel free to give feedback on this.

Comment (ArinGhazarian) [reply src/Octoshift/ReclaimService.cs:165]: yeah this is pretty much the way

File: RELEASENOTES.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
 * Improve error message when `migrate-repo` is used with a target personal access token (PAT) with insufficient permissions
 * Ensure `--no-ssl-verify` flag is honored when downloading archives from GHES
-- `--bbs-project` and `--bbs-repo` are now both required in `gh bbs2gh migrate-repo` command when `--bbs-server-url` is set
-
+* `--bbs-project` and `--bbs-repo` are now both required in `gh bbs2gh migrate-repo` command when `--bbs-server-url` is set
+* Continue to next mannequin mapping in `gh gei reclaim-mannequin --csv` if a username doesn't exist
```
Comment (ArinGhazarian) [inline RELEASENOTES.md:4]: Is there a reason you prefer start over dash? I guess the consensus is to use dash over star.

**Review by ArinGhazarian — APPROVED** [review]

File: RELEASENOTES.md (hunk 3 above)
Comment (timrogers) [reply RELEASENOTES.md:4]: I'll change them all to dash. I don't mind either way, I just wanted consistency!

**Review by dylan-smith — APPROVED** [review]

**Pull request merged**